content/setup/local-with-vagrant/_index.md

```diff
@@ -0,0 +1,198 @@
+templates and workflow targeting workers. You can check that it is running:
+
+```
+$ docker-compose ps
+        Name                      Command                  State                             Ports
+-------------------------------------------------------------------------------------------------------------------------
+deploy_boots_1         /boots -dhcp-addr 0.0.0.0: ...   Up
+deploy_cacher_1        /cacher                          Up             0.0.0.0:42111->42111/tcp, 0.0.0.0:42112->42112/tcp
+deploy_db_1            docker-entrypoint.sh postgres    Up (healthy)   0.0.0.0:5432->5432/tcp
+deploy_hegel_1         cmd/hegel                        Up
+deploy_nginx_1         /docker-entrypoint.sh ngin ...   Up             192.168.1.2:80->80/tcp
+deploy_registry_1      /entrypoint.sh /etc/docker ...   Up (healthy)
+deploy_tink-cli_1      /bin/sh -c sleep infinity        Up
+deploy_tink-server_1   tink-server                      Up (healthy)   0.0.0.0:42113->42113/tcp, 0.0.0.0:42114->42114/tcp
+```
+
+### Register the worker's hardware
+
+The provisioner is running we can follow the [example called "Hello
+world"](/examples/hello-world), I will contextualize it for our vagrant setup
+here.
+
+SSH in the provisioner again and we have to define our hello world template and load it to Tinkerbell:
+
+```
+$ cat > hello-world.yml
+version: "0.1"
+name: hello_world_workflow
+global_timeout: 600
+tasks:
+  - name: "hello world"
+    worker: "{{.device_1}}"
+    actions:
+      - name: "hello_world"
+        image: hello-world
+        timeout: 60
+$ docker exec -i deploy_tink-cli_1 tink template create --name hello-world < ./hello-world.yml
+Created Template:  75ab8483-6f42-42a9-a80d-a9f6196130df
+```
+
+Register the worker with Tinkerbell
+
+```
+$ cat > hardware-data.json
+{
+  "id": "ce2e62ed-826f-4485-a39f-a82bb74338e2",
+  "arch": "x86_64",
+  "allow_pxe": true,
+  "allow_workflow": true,
+  "facility_code": "onprem",
+  "ip_addresses": [
+    {
+      "address": "192.168.1.5",
+      "address_family": 4,
+      "enabled": true,
+      "gateway": "192.168.1.1",
+      "management": true,
+      "netmask": "255.255.255.248",
+      "public": false
+    }
+  ],
+  "network_ports": [
+    {
+      "data": {
+        "mac": "08:00:27:00:00:01"
+      },
+      "name": "eth0",
+      "type": "data"
+    }
+  ]
+}
+
+$ docker exec -i deploy_tink-cli_1 tink hardware push < ./hardware-data.json
+2020/06/17 14:12:45 Hardware data pushed successfully
+```
+
+Now that we have a template and hardware registered we can trigger the workflow for that particular hardware but first a bit of context on what we just did:
+
+1. A template is used to create workflows. Template are important because you
+   can create as many workflows you need from a single template.
+2. A workflow needs a target (a hardware) in vagrant it is the worker and it is
+   identified via MAC address.
+
+Let's start the workflow where `-t` is the template ID you got when registering
```

Comment:
This needs to be clearer - it's the only part which isn't hard-coded, so when following I copy and pasted your instructions and it didn't work for me. @gianarb